src/tests/system/tests/test_cache.py

```diff
@@ -146,66 +147,166 @@ def test_cache__writes_to_both_database_files_when_using_fully_qualified_names(
-    client.tools.getent.passwd("user-modify")
-    client.tools.getent.passwd("user-delete")
+    client.tools.getent.passwd("user1")
+    client.tools.getent.passwd("user2")
 
-    assert client.auth.ssh.password("user-modify", "Secret123"), "Login failed!"
-    assert client.auth.ssh.password("user-delete", "Secret123"), "Login failed!"
+    assert client.auth.ssh.password("user1", "Secret123"), "Login failed!"
+    assert client.auth.ssh.password("user2", "Secret123"), "Login failed!"
 
-    provider.user("user-delete").delete()
-    provider.user("user-modify").modify(shell="/bin/sh")
+    provider.user("user2").delete()
+    provider.user("user1").modify(shell="/bin/sh")
 
     client.sssctl.cache_expire(everything=True)
 
-    assert client.auth.ssh.password("user-modify", "Secret123"), "Login failed!"
-    assert not client.auth.ssh.password("user-delete", "Secret123"), "Login successful!"
+    assert client.auth.ssh.password("user1", "Secret123"), "Login failed!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "Login successful!"
 
-    result = client.tools.getent.passwd("user-modify")
+    result = client.tools.getent.passwd("user1")
     assert result is not None, "User not found!"
     assert result.shell == "/bin/sh", "User shell did not update!"
 
 
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_cache__group_entries_contains_latest_changes_when_modified_and_deleted(
+    client: Client,
+    provider: GenericProvider,
+):
+    """
+    :title: Check latest group changes are reflected when modified and deleted
+    :setup:
+        1. Create groups 'group1' and group2'
+        2. Start SSSD
+        3. Lookup groups
+    :steps:
+        1. Create and add 'user1' to 'group1' and delete 'group2' and clear cache
+        2. Lookup group 'group1'
+        3. Lookup group 'group2'
+    :expectedresults:
+        1. User 'user1' is created and added to group 'group1'  and group 'group2' is deleted
+        2. Group 'group1' is found and contains 'user1'
+        3. Group 'group2' is not found
+    :customerscenario: False
+    """
+    group1 = provider.group("group1").add()
+    group2 = provider.group("group2").add()
+    client.sssd.start()
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert client.tools.getent.group("group2"), "Group group2 not found!"
+
+    group1.add_member(provider.user("user1").add())
+    group2.delete()
+
+    client.sssctl.cache_expire(everything=True)
+
+    assert client.tools.getent.group("group1"), "Group group1 not found!"
+    assert "user1" in str(client.tools.getent.group("group1")), "Group group1 is missing a user!"
+    assert not client.tools.getent.group("group2"), "Group group2 found!"
+
+
+@pytest.mark.integration
+@pytest.mark.importance("low")
+@pytest.mark.parametrize("obj", ["user", "group", "netgroup"])
+@pytest.mark.parametrize("dbs", ["cache", "timestamps"])
```

Comment:
I understand this is more elegant way as compared to searching two DBs in a single test run.
But mind the CI overhead...